File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> result;
    void postOrder(TreeNode* node){
        if(!node) return;
        if(node->left) postOrder(node->left);
        if(node->right) postOrder(node->right);
        result.push_back(node->val);
    }

    vector<int> postorderTraversal(TreeNode* root) {
        postOrder(root);
        return result;
    }
};

File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var postorderTraversal = function(root) {
    if(!root) return [];

    const ret = [];

    const postOrder = (node) => {
        if(!node) return;

        postOrder(node.left);
        postOrder(node.right);
        ret.push(node.val);
    }

    postOrder(root);
    return ret;
};